### PR TITLE
MCOL-4282 Follow up on the previous commit.

### PR DESCRIPTION
**Pull request by tntnatbry**
Activate statement arena for:
  1. disable_indices_for_CEJ()
  2. in_subselect_rewrite() for prepared statements

### CODE REVIEW

File: dbcon/mysql/ha_mcs_pushdown.cpp [hunk 1]
```diff
@@ -826,7 +831,10 @@ create_columnstore_select_handler(THD* thd, SELECT_LEX* select_lex)
                 sel->where= conds;
 
                 if (isPS)
-                    sel->prep_where= conds;
+                {
+                    sel->prep_where= conds ? conds->copy_andor_structure(thd) : 0;
+                    in_subselect_rewrite(sel);
```
Comment (drrtuy) [inline dbcon/mysql/ha_mcs_pushdown.cpp]: We should check the return value of this in_subselect_rewrite() call and set unsupported_feature. See the block at line 859.

**Review by drrtuy — CHANGES_REQUESTED** [review]
Plz look at the comment.

File: dbcon/mysql/ha_mcs_pushdown.cpp (hunk 1 above)
Comment (tntnatbry) [reply dbcon/mysql/ha_mcs_pushdown.cpp]: Good catch. This is now fixed.

**Comment by tntnatbry** [comment]
The comment is addressed, please review the PR again.

**Review by drrtuy — APPROVED** [review]

**Pull request merged**